modules/core/shared/src/main/scala/io/circe/Printer.scala

```diff
@@ -242,12 +255,22 @@ final case class Printer(
     }
   }
 
+  @transient
+  private[this] final val stringWriter: ThreadLocal[StringBuilder] = new ThreadLocal[StringBuilder] {
+    override final def initialValue: StringBuilder = new StringBuilder()
+  }
+
   /**
    * Returns a string representation of a pretty-printed JSON value.
    */
   final def pretty(json: Json): String = {
-    val writer = new StringBuilder()
-    val folder = new PrintingFolder(writer)
+    val writer = if (reuseWriters) {
```

Comment:
I'm not sure how that would work, though—we can't reuse the array because we've returned a byte buffer wrapping it, and the user still could care about it. We could copy into the buffer but that seems to defeat the purpose to some degree…